conda/resolve.py

```diff
@@ -327,29 +327,62 @@ def filter_group(matches):
 
         # Determine all valid packages in the dependency graph
         reduced_index = {}
-        slist = list(specs)
+        explicit_spec_list = list(specs)
         for fstr in features:
             dist = Dist(fstr + '@')
             reduced_index[dist] = self.index[dist]
-        while slist:
-            this_spec = slist.pop()
-            for dist in self.find_matches(this_spec):
-                if reduced_index.get(dist) is None and self.valid(dist, filter):
-                    reduced_index[dist] = self.index[dist]
-                    for ms in self.ms_depends(dist):
-                        # We do not pull packages into the reduced index due
-                        # to a track_features dependency. Remember, a feature
-                        # specifies a "soft" dependency: it must be in the
-                        # environment, but it is not _pulled_ in. The SAT
-                        # logic doesn't do a perfect job of capturing this
-                        # behavior, but keeping these packags out of the
-                        # reduced index helps. Of course, if _another_
-                        # package pulls it in by dependency, that's fine.
-                        if 'track_features' not in ms:
-                            slist.append(ms)
+        seen_pkgs = set()
+        seen_specs = set()
+        for explicit_spec in explicit_spec_list:
+            explicit_packages = set(self.find_matches(explicit_spec)) - seen_pkgs
+            for pkg in explicit_packages:
+                seen_pkgs.add(pkg)
+                if pkg not in reduced_index and self.valid(pkg, filter):
+                    reduced_index[pkg] = self.index[pkg]
+                    dep_specs = set(self.ms_depends(pkg)) - seen_specs
```

Comment:
If a spec was added to `seen_specs` but not added to `dep_specs` because of the broadening check for a particular `this_pkg_constraints` is it safe to ignore the spec for all other packages?

I'll try to come up with a example where this would cause a false unsat result.

Reply:
Latest update nests things such that this should be addressed.

Reply:
This is equivalent to `set(self.ms_depends(pkg))` since seen_specs above